contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1301 @@
+        require(
+            _poolDistributor != address(0),
+            "Pool distributor zero address not allowed"
+        );
+
+        uint256 _tickSpacing = IMaverickV2Pool(_mPool).tickSpacing();
+        require(_tickSpacing == 1, "Unsupported tickSpacing");
+        // -1 is the tick where the contract deposits liquidity to
+        tickNumber = -1;
+
+        // tickSpacing == 1
+        (sqrtPriceTickLower, sqrtPriceTickHigher) = TickMath.tickSqrtPrices(
+            _tickSpacing,
+            tickNumber
+        );
+        sqrtPriceAtParity = _upperTickAtParity
+            ? sqrtPriceTickHigher
+            : sqrtPriceTickLower;
+
+        WETH = _wethAddress;
+        OETHp = _oethpAddress;
+        liquidityManager = IMaverickV2LiquidityManager(_liquidityManager);
+        poolLens = IMaverickV2PoolLens(_poolLens);
+        maverickPosition = IMaverickV2Position(_maverickPosition);
+        quoter = IMaverickV2Quoter(_maverickQuoter);
+        mPool = IMaverickV2Pool(_mPool);
+        votingDistributor = IVotingDistributor(_votingDistributor);
+        poolDistributor = IPoolDistributor(_poolDistributor);
+
+        require(address(mPool.tokenA()) == WETH, "WETH not TokanA");
+        require(address(mPool.tokenB()) == OETHp, "OETHp not TokanB");
+
+        // prevent implementation contract to be governed
+        _setGovernor(address(0));
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     */
+    function initialize() external onlyGovernor initializer {
+        // Read reward
+        address[] memory _rewardTokens = new address[](1);
+        _rewardTokens[0] = poolDistributor.rewardToken();
+
+        require(_rewardTokens[0] != address(0), "No reward token configured");
+
+        InitializableAbstractStrategy._initialize(
+            _rewardTokens,
+            new address[](0),
+            new address[](0)
+        );
+    }
+
+    /***************************************
+                  Configuration 
+    ****************************************/
+
+    /**
+     * @notice Set allowed pool weth share interval. After the rebalance happens
+     * the share of WETH token in the ticker needs to be withing the specifications
+     * of the interval.
+     *
+     * @param _allowedWethShareStart Start of WETH share interval expressed as 18 decimal amount
+     * @param _allowedWethShareEnd End of WETH share interval expressed as 18 decimal amount
+     */
+    function setAllowedPoolWethShareInterval(
+        uint256 _allowedWethShareStart,
+        uint256 _allowedWethShareEnd
+    ) external onlyGovernor {
+        require(
+            _allowedWethShareStart < _allowedWethShareEnd,
+            "Invalid interval"
+        );
+        // can not go below 1% weth share
+        require(_allowedWethShareStart > 0.01 ether, "Invalid interval start");
+        // can not go above 95% weth share
+        require(_allowedWethShareEnd < 0.95 ether, "Invalid interval end");
```

Comment:
nit: Maybe we could replace magic numbers (`0.01 ether` and `0.95 ether`) by a constant/immutable with an intuitive name?

Reply:
If a constant number is only used once and is used in a numeric sense, I often prefer it to live where it is used. This makes reviewing and understanding the code easier.

If however, it actually represents something that's not a number, then it's good to constant it up.

For example, if a "2" is actually "MODE_EXACT_INPUT_IN", it really helps to make it a constant. But in this, case, "don't allow the range to be out of 95%" is best have it clear in the code.